tests/test_bgbb.py

```diff
@@ -69,15 +76,48 @@ def test_expec_p_th(bg):
     sim_expec_p_th_diffs(bg, abgd=[70.0, 30.0, 25.0, 75.0], seed=0)
 
 
+def test_p_interval(bgdat):
+    p0n1 = bgdat.rfn.p_x_interval(n_star=1, x_star=0)
+    p0n2 = bgdat.rfn.p_x_interval(n_star=2, x_star=0)
+
+    assert check_prob(p0n1)
+    assert check_prob(p0n2)
+
+    p_any_n1 = 1 - p0n1
+    p_any_n2 = 1 - p0n2
+    assert (p_any_n2 > p_any_n1).all()
+
+
+def test_p_interval_close_p_alive(bgdat, df):
+    """
+    The threshold for this test is somewhat arbitrary,
+    but should check that P(ever return in a long interval)
+    is a close approximation of P(alive).
+    """
+
+    p_alive = bgdat.rfn.cond_prob_alive(df)
+
+    p_never_return = bgdat.rfn.p_x_interval(n_star=100, x_star=0)
+    p_ever_return = 1 - p_never_return
+
+    assert check_prob(p_alive)
+    assert check_prob(p_never_return)
+    assert check_prob(p_ever_return)
+
+    assert np.corrcoef(p_alive, p_ever_return)[0, 1] > .99
+
+    # Test case for x_star != 0
+    p_return_once = bgdat.rfn.p_x_interval(n_star=100, x_star=1)
+    assert check_prob(p_return_once)
```

Comment:
Is `p_return_once` also close to `p_alive`, as per the name of the test?

Reply:
So if they're alive, the probability is high that they'll return between 1 times-n times where n is a large number. For a large interval (say a month) the probability is pretty low that you'll return *exactly* once. But you can get the probability that they'll return any non-zero # times in the interval by taking 1 - P(return exactly 0 times). And I'm assuming this is a good proxy for p_alive. Hopefully make sense.